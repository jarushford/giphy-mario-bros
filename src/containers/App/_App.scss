@font-face {
  font-family: 'Mario';
  src: url('./assets/SuperMario256.ttf') format('truetype');
}

.App {
  width: 100%;
  height: 96vh;
  background: url('./assets/bg.jpg');
  background-size: cover;
  padding-top: 30px;
}