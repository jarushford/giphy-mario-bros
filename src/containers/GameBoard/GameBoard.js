import React from 'react'
import Loading from '../../components/Loading/Loading'
import PlayerScore from '../../components/PlayerScore/PlayerScore'
import { connect } from 'react-redux'
import { uid } from 'react-uid';
import '../../main.scss'

export function GameBoard({ players, judge, caption, currentPlayer }) {
  if (!players.length) {
    return <Loading />
  }

  const scores = players.map(player => {
    if (player.player === judge) {
      return <PlayerScore {...player} key={uid(player)} active={true} />
    }
    return <PlayerScore {...player} key={uid(player)} active={false} />
  })

  let instruction
  if (currentPlayer === judge) {
    instruction = <h2>Player {currentPlayer}, choose a winner</h2>
  } else {
    instruction = <h2>Player {currentPlayer}, choose a GIF</h2>
  }

  return (
    <section className="game-board">
      <h1 className="title">Giphy Mario Bros.</h1>
      <div className="player-scores">{scores}</div>
      <h3>{caption[0]}</h3>
      {instruction}  
    </section>
  )
}

export const mapStateToProps = (state) => ({
  players: state.players,
  judge: state.judge,
  currentPlayer: state.currentPlayer,
  caption: state.caption
})

export default connect(mapStateToProps)(GameBoard)