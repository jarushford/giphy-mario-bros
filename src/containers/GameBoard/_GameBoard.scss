.game-board {
  @include main();
  cursor: default;
  min-width: 700px;
  margin-top: 70px;

  h3 {
    text-align: center;
    color: $white;
    background: $red;
    padding: 10px;
    border-radius: 4px;
    font-family: 'Viga', sans-serif;
    margin: 10px auto 30px;
  }

  h2 {
    text-align: center;
    color: $white;
    font-family: 'Mario';
    text-shadow: $outlinethin;
    margin: 25px auto 5px;
  }
}

.player-scores {
  display: flex;
  justify-content: space-around;
}