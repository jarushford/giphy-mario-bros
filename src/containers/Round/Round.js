import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import '../../main.scss'

export function Round({ round, players }) {
  if (!players.length) {
    return <div />
  }

  return (
    <div className="round">
      <h3 className="round-num">{round + 1}</h3>
      <h3 className="round-sep">/</h3>
      <h3 className="round-total">{players.length * 2 + 1}</h3>
    </div>
  )
}

export const mapStateToProps = (state) => ({
  round: state.round,
  players: state.players
})

export default connect(mapStateToProps)(Round)

Round.propTypes = {
  round: PropTypes.number.isRequired,
  players: PropTypes.array.isRequired
}