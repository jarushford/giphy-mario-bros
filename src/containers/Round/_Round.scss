.round {
  position: absolute;
  top: 0px;
  left: 20px;
  display: flex;
  color: rgb(233, 12, 12);
  text-shadow: $outlinethin;
  font-family: 'Mario';
  font-size: 2rem;
}

.round-num {
  position: relative;
  bottom: 10px;
}

.round-total {
  position: relative;
  top: 10px;
}