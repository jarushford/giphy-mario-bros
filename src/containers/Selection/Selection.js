import React, { Component } from 'react'
import { connect } from 'react-redux'
import { uid } from 'react-uid';

export class Selection extends Component {
  constructor() {
    super()
    this.state = {
      adjustment: 0
    }
  }

  adjustCarousel = (adj) => {
    const { adjustment}  = this.state
    if (adjustment < 4 && adj === 1) {
      this.setState({ adjustment: this.state.adjustment + adj })
    } else if (adjustment > 0 && adj === -1) {
      this.setState({ adjustment: this.state.adjustment + adj })
    }
  }

  render() {
    const { players, currentPlayer } = this.props
    const { adjustment } = this.state
    const style = { transform: `translateX(${adjustment * -290}px)` }
    let gifs = []
    if (players.length) {
      gifs = players[currentPlayer].gifs.map(gif => {
        return <img
          className="gif"
          src={gif.images.fixed_width.url}
          key={uid(gif)}
          alt=''
          style={style}  
        />
      })
    }

    return (
      <section className="selection">
        <button onClick={() => this.adjustCarousel(-1)}>
          <i class="fas fa-arrow-left" />
        </button>
        <div className="gifs-container">
          {gifs}
        </div>
        <button onClick={() => this.adjustCarousel(1)}>
          <i class="fas fa-arrow-right" />
        </button>
      </section>
    )
  }
}

export const mapStateToProps = (state) => ({
  players: state.players,
  currentPlayer: state.currentPlayer
})

export default connect(mapStateToProps)(Selection)