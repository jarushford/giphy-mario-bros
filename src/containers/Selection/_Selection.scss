.selection {
  display: flex;
  justify-content: space-around;
  align-items: center;
  margin-top: 40px;

  button {
    @include btn();
    height: 50px;
    width: 50px;
  }
}

.gifs-container {
  display: grid;
  width: 80%;
  overflow: hidden;
  margin: 0 26px 0 8px;
}

.gif {
  box-sizing: border-box;
  grid-row: 1;
  width: 250px;
  height: 200px;
  margin: 20px;
  transition: .3s ease;
  border: 5px solid $bg;
  border-radius: 4px;
  cursor: pointer;

  &:hover {
    border: 5px solid #ffdf30;
  }
}

.judge-choice {
  margin: 0 30px;
  max-width: 700px;
  grid-gap: 0px 30px;
}

.judge-choice .gif {
  margin: 20px auto;
}

.fa-arrow-left,
.fa-arrow-right {
  padding-top: 7px;
}

.start-turn {
  @include btn();
  margin-top: 80px;
  width: 250px;
}