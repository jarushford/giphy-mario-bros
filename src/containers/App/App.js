import React, { Component } from 'react'
import { Route, Switch, Link, withRouter } from 'react-router-dom'
import Welcome from '../../components/Welcome/Welcome'
import SelectPlayers from '../SelectPlayers/SelectPlayers'
import GameBoard from '../GameBoard/GameBoard'
import Selection from '../Selection/Selection'
import Round from '../Round/Round'
import NewGame from '../NewGame/NewGame'
import Error from '../Error/Error'
import { connect } from 'react-redux'
import '../../main.scss'

export class App extends Component {
  render() {
    if (this.props.error) {
      return <div className="App">
        <Error />
      </div>
    }

    return (
      <div className="App">
        <Switch>
          <Route exact path='/' component={Welcome} />
          <Route exact path='/select-players' component={SelectPlayers} />
          <Route exact path='/home' render={() => {
            return (
              <div>
                <Round />
                <Link to='/'><button className="home"><i className="fa fa-home"/></button></Link>
                <GameBoard />
                <Selection />
              </div>
            )
          }} />
          <Route exact path='/newgame' component={NewGame} />
          <Route component={Error}/>
        </Switch>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  error: state.error
})

export default withRouter(connect(mapStateToProps)(App))
