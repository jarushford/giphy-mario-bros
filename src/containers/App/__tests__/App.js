import React from 'react'
import { App, mapStateToProps } from '../App'
import { shallow, mount } from 'enzyme'
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import { MemoryRouter, Link } from 'react-router-dom'
import Round from '../../Round/Round'
import GameBoard from '../../GameBoard/GameBoard'
import Selection from '../../Selection/Selection'

describe('App', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<App />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should match the snapshot with an error', () => {
    const wrapper = shallow(<App error={'error'} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should return a props object from mapStateToProps', () => {
    const mockState = {
      error: '',
      judge: 1
    }
    const expected = { error: '' }

    const mappedProps = mapStateToProps(mockState)
    expect(mappedProps).toEqual(expected)
  })

  it('should render correct components if on home route', () => {
    const wrapper = mount(
      <Provider store={createStore(() => ({
        round: 1,
        players: [],
        choices: [],
        usedCaptions: [],
        currentPlayer: 1,
        judge: 3,
        caption: [],
        unusedIDs: []
      }))}>
        <MemoryRouter initialEntries={['/home']}>
          <App />
        </MemoryRouter>
      </Provider>
    )
    expect(wrapper.find(Round)).toHaveLength(1)
    expect(wrapper.find(GameBoard)).toHaveLength(1)
    expect(wrapper.find(Selection)).toHaveLength(1)
    expect(wrapper.find(Link)).toHaveLength(1)
  })
})