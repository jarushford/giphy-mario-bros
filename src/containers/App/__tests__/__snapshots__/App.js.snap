// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/select-players"
        />
        <Route
          exact={true}
          path="/home"
          render={[Function]}
        />
        <Route
          component={[Function]}
          exact={true}
          path="/newgame"
        />
        <Route
          component={[Function]}
        />
      </Switch>,
      "className": "App",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/select-players"
          />,
          <Route
            exact={true}
            path="/home"
            render={[Function]}
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/newgame"
          />,
          <Route
            component={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/select-players",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/home",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/newgame",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/select-players"
          />
          <Route
            exact={true}
            path="/home"
            render={[Function]}
          />
          <Route
            component={[Function]}
            exact={true}
            path="/newgame"
          />
          <Route
            component={[Function]}
          />
        </Switch>,
        "className": "App",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/select-players"
            />,
            <Route
              exact={true}
              path="/home"
              render={[Function]}
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/newgame"
            />,
            <Route
              component={[Function]}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/select-players",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/home",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/newgame",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`App should match the snapshot with an error 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    error="error"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Connect(Error) />,
      "className": "App",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Connect(Error) />,
        "className": "App",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
